src/raft_log.rs

```diff
@@ -1185,6 +1202,91 @@ mod test {
                 );
             }
         }
+
+        let ents = [
```

Comment:
Do we need a case that applied > last index after restart, and make sure the init process fails?

Reply:
It is possible applied > last_index and the it won't let init process fail. Call `next_entries` will just return None. Test case added.

Reply:
I change to code to let initialization not call `applied_to` so we can keep the check.